src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -633,7 +633,7 @@ void GCToOSInterface::YieldThread(uint32_t switchCount)
 static void* VirtualReserveInner(size_t size, size_t alignment, uint32_t flags, uint32_t hugePagesFlag = 0)
 {
     assert(!(flags & VirtualReserveFlags::WriteWatch) && "WriteWatch not supported on Unix");
-    if (alignment == 0)
+    if (alignment < OS_PAGE_SIZE)
```

Comment:
Wrap under `#if defined(HOST_ARM64) && defined(TARGET_OSX) && defined(FEATURE_NATIVEAOT)` to limit the scope of this change?

Reply:
I would still like to know why CoreCLR doesn't hit this but I assume that it's because of some GC feature switch (such as `FEATURE_MANUALLY_MANAGED_CARD_BUNDLES`).

While I can add the `#if` guards the code would still misbehave fatally when `alignment < OS_PAGE_SIZE` so I think the change is quite safe.

Reply:
> I would still like to know why CoreCLR doesn't hit this

AFAIK gcenv.unix.cpp is only used by CoreCLR when standalone GC is used (i.e. COMPlus_GCName is set to point to a standalone GC instead of the one in coreclr.dll/.so). @mangod9 @dotnet/gc can you have a look at this change?

Reply:
> AFAIK, there is no azp run trigger for that pipeline, someone with access will need to trigger it manually.

I tried to trigger it but it's complaining about the pool. It doesn't look like the gc-standalone pipeline ran for a week or so. It coincides with the pool changes. Cc @dotnet/gc.

The interesting thing about the `gc-standalone` pipeline is that it doesn't test mac at all. So if there was a arm64 macOS issue with standalone GC, we wouldn't even know it.

Reply:
> if there was a arm64 macOS issue with standalone GC, we wouldn't even know it.

This change as written is not specific to arm64 macOS, other platforms can break. My original commit was https://github.com/am11/runtime/commit/45f3a4844126da5013980873af76fc3bd06e01f1, but it was recreated without the guard which has potential to break existing code.

Reply:
The specific call with the "invalid" `alignment` parameter is macOS arm64 specific though. It happens because of 16Kb page sizes and `GC_PAGE_SIZE != OS_PAGE_SIZE`.